linode/manager
# feat: [M3-7043] - Improve layout of support ticket breadcrumb and status details

## Description 📝
This PR makes changes to the existing breadcrumb on a support ticket's details page.

The problems:
- Stacked label looks busy and does not fit with our visual language
- Chip does not match component library
- Layout of "last updated" information - it does not fit well within the breadcrumb

## Changes  🔄
- Replaces the status `Chip` with a `StatusIcon`.
   - After a discussion with UX, we decided to go with grey (`inactive`) for the icon color, rather than red, to avoid implying that there's an error status.
- Removes the status details from the breadcrumb.
   - It doesn't format nicely and I'd argue that that information doesn't belong in a breadcrumb, anyway. This data is specific to the support ticket details page.
- Removes the notice from the support ticket details page, since the entity is included in the new status details layout.
- Cleans up the Breadcrumb component(s) to remove the `subtitle` prop from the `labelOptions`. 
- Improves test coverage for the changes in `SupportTicketDetail.test.tsx`.
- Converted StatusIcon story from mdx to tsx.

## Preview 📷
| Before  | After   |
| ------- | ------- |
|![Screenshot 2023-10-31 at 8 40 23 AM](https://github.com/linode/manager/assets/114685994/bb9d10fa-16ef-4868-ae67-39877fd73e12) |![Screenshot 2023-10-31 at 10 18 21 AM](https://github.com/linode/manager/assets/114685994/885ac8bf-b38c-4dc4-858b-aeda1818792e) |
| ![Screenshot 2023-10-31 at 1 33 46 PM](https://github.com/linode/manager/assets/114685994/629c7e4d-a7d3-4d41-8510-44b776793acb) | ![Screenshot 2023-10-31 at 1 33 23 PM](https://github.com/linode/manager/assets/114685994/0a52f015-335d-42dd-b57e-b92ab60fc489) |
| ![Screenshot 2023-10-31 at 1 34 07 PM](https://github.com/linode/manager/assets/114685994/2f1ec2bf-b663-44d7-b704-66ec05553028) | ![Screenshot 2023-10-31 at 1 32 40 PM](https://github.com/linode/manager/assets/114685994/efc5b2aa-6d97-4aa5-a391-25ad6424919a)|

## How to test 🧪

### Prerequisites
(How to setup test environment)
- Check out this PR.
- **If you want to test this by actually creating a new support ticket and then looking at it in the UI, please set your environment to `dev` to avoid bothering Support.**

### Reproduction steps
(How to reproduce the issue, if applicable)
- Go to https://cloud.linode.com/support/tickets.
- Look at any closed tickets on your account. Observe the odd layout of the chip and the subtitle under the breadcrumb.
   - If you have none, check out `develop` and mock some using the MSW.
   - Edit the endpoint in `serverHandlers.ts`: `rest.get('*/support/tickets/999'` and add `status: closed`.
### Verification steps 
(How to verify changes)
- Go to http://localhost:3000/support/tickets.
- Look at the tickets on your account and confirm:
   - Status is either `New`, `Open`, or `Closed`.
   - When the status is new or open, the status icon is green. When the status is closed, the icon is grey.
   - If the ticket has an associated entity, the entity should be listed as "Regarding: ..." and, if applicable, a link to that entity. Otherwise, that text should not display. 
   - You may have to mock tickets and turn on the MSW if you don't have any with the right `status` or `entity` included. (see the endpoint above to edit)
- Confirm unit test passes.

```
yarn test SupportTicketDetail
```
- Confirm StatusIcon story displays in Storybook: http://localhost:6006/?path=/docs/components-statusicon--documentation
<details>
<summary>
Storybook 📸 
</summary>

![Screenshot 2023-11-03 at 3 21 48 PM](https://github.com/linode/manager/assets/114685994/991edd28-e394-48a2-8f01-9dfa177f6b07)
</details>

## As an Author I have considered 🤔

*Check all that apply*
